lib/api/BackbeatAPI.js

```diff
@@ -1336,26 +1307,70 @@ class BackbeatAPI {
      */
     _setupMongoClient(cb) {
         const mongoConfig = this._config.queuePopulator.mongo;
-        const mongoUrl = constructConnectionString(mongoConfig);
-        MongoClient.connect(mongoUrl, {
+        if (!mongoConfig) {
+            this._logger.debug('MongoDB configuration not found, skipping MongoDB client setup', {
+                method: 'BackbeatAPI._setupMongoClient',
+            });
+            return cb();
+        } else {
+            const mongoUrl = constructConnectionString(mongoConfig);
+            return MongoClient.connect(mongoUrl, {
                 replicaSet: mongoConfig.replicaSet,
                 useNewUrlParser: true,
                 useUnifiedTopology: true,
```

Comment:
Not related to the changes, but I believe this is not used anymore with recent mongodb drivers, but I also see that we are still using mongo driver v3 in Backbeat... Would it make sense to bump it in this PR, as we modify this function? or we can do it when upgrading the deps (but these lines will be easy to miss...)

Reply:
I suggest adding a comment so it is not missed during the dependencies bump